ansible/roles/gpu/tasks/nvidia-gpu.yaml

```diff
@@ -2,18 +2,115 @@
 - name: "NVIDIA Gpu - CentOS 7"
   include_tasks: nvidia-gpu-CentOS7.yaml
   when: ansible_distribution == "CentOS" and ansible_distribution_major_version == '7'
+
 - name: "NVIDIA Gpu - RHEL 7"
   include_tasks: nvidia-gpu-RHEL7.yaml
   when: ansible_distribution == 'RedHat' and ansible_distribution_major_version == '7'
+
 - name: "NVIDIA Gpu - RHEL 8"
   include_tasks: nvidia-gpu-RHEL8.yaml
   when: ansible_distribution == 'RedHat' and ansible_distribution_major_version == '8'
+
 - name: "NVIDIA Gpu - SLES 15"
   include_tasks: nvidia-gpu-SLES15.yaml
   when: ansible_distribution == "SLES" and ansible_distribution_major_version == '15'
+
 - name: "NVIDIA Gpu - Flatcar"
   include_tasks: nvidia-gpu-Flatcar.yaml
   when: ansible_distribution == "Flatcar"
+
 - name: "NVIDIA Gpu - Ubuntu 20"
   include_tasks: nvidia-gpu-Ubuntu20.yaml
   when: ansible_distribution == "Ubuntu"
+
+# disable with these instructions https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#runfile-nouveau
+- name: create nouveau blacklist file
+  copy:
+    src: ./blacklist-nouveau.conf
+    dest: /etc/modprobe.d/blacklist-nouveau.conf
+
+- name: regenerate the kernel initramfs
+  command: dracut --force
+  become: yes
+  when:
+    - ansible_os_family == "RedHat"
+
+- name: regenerate the kernel initramfs
+  command: update-initramfs -u
+  become: yes
+  when: ansible_os_family == "Debian"
+
+# to pick up the changes to unload nouveau
+# some hardware like aws g4dn.2xlarge require this
+- name: unconditionally reboot the machine with all defaults
+  ansible.builtin.reboot:
+  when: sysprep
+
+# Download and install gpu runfile
+- name: create nvidia install directory
+  file:
+    path: /opt/dkp/nvidia
+    state: directory
+
+- name: download gpu runfile
+  get_url:
+    url: "{{ nvidia_runfile_installer_url }}"
+    dest: "{{ nvidia_remote_bundle_path }}/{{ nvidia_runfile_installer }}"
+    mode: 711
+  when:
+   - not offline_mode_enabled
+
+- name: extract driver source files
+  shell: |
+    "{{ nvidia_remote_bundle_path }}/{{ nvidia_runfile_installer }}" -x -s
+  args:
+    chdir: /opt/dkp/nvidia
+  become: yes
+
+- name: install driver
+  shell:
+    ./nvidia-installer -s \
+      --log-file-name=$(pwd)/nvidia-installer.log || true
```

Comment:
should not ignore exit code for driver installation failures

Reply:
the test we're using is `nvidia-smi` to determine whether or not the installation works let me try removing and see what it does. the installer can sometimes give a false negative which is why i added the `|| true` there